### ISSUE TITLE
False positive in `Lint/RedundantSafeNavigation` (regression)

### ISSUE THREAD

**Issue by vlad-pisanov** [rubocop/rubocop#15070]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Lint/RedundantSafeNavigation

### Steps to reproduce the problem

Consider
```ruby
user = find_user(123)
unless user&.address&.street
  puts 'no street'
end
```

### Expected behavior

no offence since `user` can be `nil`

### Actual behavior

in `1.85.1` this is valid

in `1.86.0`, `Lint/RedundantSafeNavigation` incorrectly flags `unless user&.address&.street` as an offense claiming that `user` is non-nil.
```
Lint/RedundantSafeNavigation: Redundant safe navigation on non-nil receiver (detected by analyzing previous code/method invocations).
unless user&.address&.street
           ^^
```

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
1.86.0 (using Parser 3.3.11.1, Prism 1.9.0, rubocop-ast 1.49.1, analyzing as Ruby 3.3, running on ruby 3.3.2) [x86_64-darwin24]
  - rubocop-minitest 0.39.1
  - rubocop-performance 1.26.1
  - rubocop-rails 2.34.3
  - rubocop-factory_bot 2.28.0
```
